src/main/java/com/teragrep/jos_01/procfs/status/Cpuinfo.java

```diff
@@ -0,0 +1,381 @@
+
+    public Cpuinfo(ArrayList<String> rows) {
+        this.rows = rows;
+        this.processors = new ArrayList<Processor>();
+        this.timestamp = LocalDateTime.now();
+        this.statistics = new LinkedHashMap<>();
+
+        HashMap<String, String> processorFields = new LinkedHashMap<String, String>();
+        for (String row : rows) {
+            if (row.equals("")) {
+                Processor processor = new Processor(
+                        processorFields.get("processor"),
+                        processorFields.get("vendor id"),
+                        processorFields.get("cpu family"),
+                        processorFields.get("model"),
+                        processorFields.get("model name"),
+                        processorFields.get("stepping"),
+                        processorFields.get("microcode"),
+                        processorFields.get("cpu MHz"),
+                        processorFields.get("cache size"),
+                        processorFields.get("physical id"),
+                        processorFields.get("siblings"),
+                        processorFields.get("core id"),
+                        processorFields.get("cpu cores"),
+                        processorFields.get("apicid"),
+                        processorFields.get("initial apicid"),
+                        processorFields.get("fpu"),
+                        processorFields.get("fpu_exception"),
+                        processorFields.get("cpuid level"),
+                        processorFields.get("wp"),
+                        processorFields.get("flags"),
+                        processorFields.get("vmx flags"),
+                        processorFields.get("bugs"),
+                        processorFields.get("bogomips"),
+                        processorFields.get("clflush size"),
+                        processorFields.get("cache_alignment"),
+                        processorFields.get("address sizes"),
+                        processorFields.get("power management"),
+                        processorFields
+                );
+                processors.add(processor);
+                processorFields = new LinkedHashMap<String, String>();
+                continue;
+            }
+            String[] keyValuePair = row.split(":");
+            String key = keyValuePair[0].trim();
+            String value;
+            if (keyValuePair.length == 1) {
+                value = "";
+            }
+            else {
+                value = keyValuePair[1].trim();
+            }
+            processorFields.put(key, value);
+        }
+    }
+
+    public ArrayList<Processor> processors() {
+        return processors;
+    }
+
+    public int cpuPhysicalCoreCount() {
+        Map<Integer, Integer> coreCounts = new HashMap<>();
+        for (Processor processor : processors) {
+            coreCounts.put(processor.physical_id, processor.cpu_cores);
+        }
+        int coreCount = 0;
```

Comment:
I think this function could be optimized down to
```
int coreCount = 0;
for(Processor processor : processors) {
    coreCount += proecssor.cpu_cores
}
return coreCount;
```

Reply:
This could lead to duplication, because cpuinfo's processors are logical processors, and each reports the number of physical cpu cores the associated physical processor has.

The first loop goes through every logical processor, and adds each physical processor's cpu_cores. Since the keys in Map have to be unique, adding a cpu_cores value to an already existing key just updates that key's value instead of adding them together, handling the duplication issue.

We could just read the first occurrence of cpu_cores and call it a day, but it would be incorrect if the system has multiple physical CPU's installed.

Perhaps Cpuinfo.Processor should be renamed to Cpuinfo.LogicalProcessor

Reply:
Sounds reasonable